Python/Python-tokenize.c

```diff
@@ -302,7 +330,9 @@ tokenizeriter_next(tokenizeriterobject *it)
 exit:
     _PyToken_Free(&token);
     if (type == ENDMARKER) {
+        Py_BEGIN_CRITICAL_SECTION(it);
         it->done = 1;
```

Comment:
You could change this to an atomic, which might be faster than locking? See the `FT_ATOMIC_STORE_X` macros.